server/pulp/server/db/querysets.py

```diff
@@ -116,6 +119,28 @@ def get_or_404(self, **kwargs):
             raise pulp_exceptions.MissingResource(**kwargs)
 
 
+class WorkerQuerySet(CriteriaQuerySet):
+    """
+    Custom queryset for workers
+    """
+
+    def get_online(self):
+        """
+        Returns a queryset with a subset of Worker documents.
+
+        The queryset is filtered to remove any Worker document that has not been updated in the
+        last 25 seconds.
+
+        :return: mongoengine queryset object
+        :rtype:  mongoengine.queryset.QuerySet
+        """
+        query_set = self
+        now = ensure_tz(datetime.utcnow())
+        oldest_heartbeat_time = now - timedelta(seconds=constants.PULP_PROCESS_TIMEOUT_INTERVAL)
+        query_set.filter(last_heartbeat__gte=oldest_heartbeat_time)
```

Comment:
The return value of this statement needs to be returned. As it's written now the original queryset is returned. Remove L#141 and make this `return query_set.filter(last_heartbeat__gte=oldest_heartbeat_time)`